### DISCUSSION TITLE
Svelte Grammar is outdated for usage with Typescript

### DISCUSSION THREAD

**Discussion by sebastinez** [github-linguist/linguist#6215] [Classification]
Currently linguist uses https://github.com/umanghome/svelte-atom as grammar for svelte.
The issue is that Svelte has a somewhat new typescript directive `lang="ts"` in the `script` tag, which can create issues when rendering.

See here https://github.com/radicle-dev/radicle-interface/blob/master/src/Authorship.svelte

The correct way to highlight this should be:
![localhost_3000_seeds_clients radicle xyz_rad_git_hnrkjajuucc6zp5eknt3s9xykqsrus44cjimy_tree_cfb1c5acd103772842bf6465359156476ce4e892_src_Authorship svelte](https://user-images.githubusercontent.com/7912302/208061351-3337897e-8073-4e82-89af-2e82a343cf8c.png)

I would suggest while waiting for the tree-sitter-svelte grammar to be solid, to have a new updated fork for the pretty lights svelte grammar.

I have updated the grammar on a fork of mine in the meantime https://github.com/sebastinez/svelte-atom/tree/sebastinez/add-lang-ts

**Comment by lildude** [comment]
We can certainly switch to using your fork, but before we do, you might want to check on the progress of the https://github.com/sveltejs/svelte-atom grammar. We switched to it but had to revert as it broke the syntax highlighting of Svelte code between `<(script|style|template)>` tags. See https://github.com/github/linguist/discussions/6164 for the full discussion and history and upstream issue links.

**Comment by sebastinez** [comment]
Hey @lildude,

Thanks for your quick response! I already went down this rabbit hole 😅 
It seems that the `sveltejs/svelte-atom` didn't had any improvements in the last months, since you switched to it and then reverted back from it.
Regarding the fix I have in my fork over the one from @umanghome it seems that this is already addressed in the svelte-atom repo https://github.com/sveltejs/svelte-atom/blob/865a55f096ed0d60a316234f55f6f356abd5cd1f/grammars/svelte.cson#L42

Have to compile the svelte-atom grammar to json to see what's still broken there..

**Reply by lildude** [reply]
> Regarding the fix I have in my fork over the one from @umanghome it seems that this is already addressed in the svelte-atom repo https://github.com/sveltejs/svelte-atom/blob/865a55f096ed0d60a316234f55f6f356abd5cd1f/grammars/svelte.cson#L42
> 
> Have to compile the svelte-atom grammar to json to see what's still broken there..

No need. Linguist can use CSON grammars. The problem is the fix you've pointed to won't work until the grammar is updated to address the `$` naming problem identified in https://github.com/github/linguist/discussions/6164#discussioncomment-4178987

**Reply by sebastinez** [reply]
Ok from what I can see, the issue that started the entire discussion you pointed at is here https://github.com/sveltejs/svelte-atom/issues/20 and has to do with the `template` tag not being handled correctly, which ended in reverting the usage of the `sveltejs/svelte-atom` repo.

So AFAIC see the `umanghome/svelte-atom` repo has this issue handled, but since it's not maintained lost some of the newer svelte features.
So I'll create a PR to use my fork of the `umanghome/svelte-atom` repo where the newest features are implemented and the `$` naming problem isn't an issue, since we declare everything explicitly.

If at any point in time the svelte treesitter grammar improves enough to be used instead this should be done of course.

**Comment by sebastinez** [comment]
Ok I wrote a small PR that moves the Svelte Grammar definition to a maintained fork
https://github.com/github/linguist/pull/6216
